src/transformers/modeling_utils.py

```diff
@@ -605,14 +605,13 @@ def resize_token_embeddings(self, new_num_tokens: Optional[int] = None) -> torch
         Return:
             :obj:`torch.nn.Embedding`: Pointer to the input tokens Embeddings Module of the model.
         """
-        base_model = getattr(self, self.base_model_prefix, self)  # get the base model if needed
```

Comment:
I thought about this a lot and I really don't get why this function is here. It doesn't make much sense to me and should be deleted.
- There could be a case when the LMHead model has no `get_input_embeddings()` function, but this is already checked in the `get_input_embeddings()` function of this class. Same for `set_input_embeddings()` -> thus this line doesn't help here at all and would only break the `get_output_embeddings()` function. 

 Maybe @LysandreJik @sgugger @thomwolf can double-check here though...

Reply:
I don't have enough context here to comment, so unless there is a deeper reason, I'd go with the simpler code as long as the tests pass.

Reply:
Yes, that's redundant.